src/qml/bitcoinamount.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2024 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <qml/bitcoinamount.h>
+
+#include <QRegExp>
+#include <QStringList>
+
+
+BitcoinAmount::BitcoinAmount(QObject *parent) : QObject(parent)
+{
+    m_unit = Unit::BTC;
+}
+
+int BitcoinAmount::decimals(Unit unit)
+{
+    switch (unit) {
+    case Unit::BTC: return 8;
+    case Unit::SAT: return 0;
+    } // no default case, so the compiler can warn about missing cases
+    assert(false);
+}
+
+QString BitcoinAmount::sanitize(const QString &text)
+{
+    QString result = text;
+
+    // Remove any invalid characters
```

Comment:
Is this the right thing to do?
QT gui simply refuses to paste when there are invalid characters.